src/js/net.js

```diff
@@ -279,7 +279,11 @@ function connect(socket, ip, port) {
     }
   };
 
-  socket._handle.connect(ip, port, afterConnect);
+  var err = socket._handle.connect(ip, port, afterConnect);
+  if (err) {
+    emitError(socket, new Error('connect failed - status: ' +
+      TCP.errname(err)));
+  }
```

Comment:
I think the function `emitError` is the function to emit error instantly and I think the meaning `it is done at the next tick` is not meant. If you want to try to do that, you have to do yourself like below I think.
```
if (err) {
  process.nextTick(function() {
    socket.emit('error',  new Error('connect failed - status: ' + TCP.errname(err)));
  }
}
```